File: frontend/lib/features/approval/domain/entities/recommendation.dart
import 'package:equatable/equatable.dart';

enum RecommendationType {
  approve,
  review,
  reject,
}

class Recommendation extends Equatable {
  final String id;
  final String packageId;
  final RecommendationType type;
  final String evidence;
  final DateTime createdAt;

  const Recommendation({
    required this.id,
    required this.packageId,
    required this.type,
    required this.evidence,
    required this.createdAt,
  });

  @override
  List<Object?> get props => [id, packageId, type, evidence, createdAt];
}

File: backend/src/BajajDocumentProcessing.Domain/Entities/Recommendation.cs
using BajajDocumentProcessing.Domain.Common;
using BajajDocumentProcessing.Domain.Enums;

namespace BajajDocumentProcessing.Domain.Entities;

/// <summary>
/// Represents an AI-generated recommendation for document package approval, rejection, or further review.
/// Includes evidence and validation issues to support the recommendation
/// </summary>
public class Recommendation : BaseEntity
{
    /// <summary>
    /// Gets or sets the unique identifier of the document package this recommendation is for
    /// </summary>
    public Guid PackageId { get; set; }
    
    /// <summary>
    /// Gets or sets the type of recommendation (Approve, Reject, RequestMoreInfo, FlagForReview)
    /// </summary>
    public RecommendationType Type { get; set; }
    
    /// <summary>
    /// Gets or sets the AI-generated evidence supporting this recommendation
    /// </summary>
    public string Evidence { get; set; } = string.Empty;
    
    /// <summary>
    /// Gets or sets the JSON representation of validation issues found during analysis
    /// </summary>
    public string? ValidationIssuesJson { get; set; }
    
    /// <summary>
    /// Gets or sets the overall confidence score (0-100) for this recommendation
    /// </summary>
    public double ConfidenceScore { get; set; }

    /// <summary>
    /// Gets or sets the document package this recommendation is for
    /// </summary>
    public DocumentPackage Package { get; set; } = null!;
}
